var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -296,8 +297,11 @@ class Curl(NMakePackage, AutotoolsPackage):
     conflicts("platform=linux", when="tls=secure_transport", msg="Only supported on macOS")
 
     depends_on("gnutls", when="tls=gnutls")
-    depends_on("mbedtls@2: +pic", when="@7.79: tls=mbedtls")
-    depends_on("mbedtls@:2 +pic", when="@:7.78 tls=mbedtls")
+    with when("tls=mbedtls"):
+        depends_on("mbedtls +pic")
+        depends_on("mbedtls@:2", when="@:7.78")
+        depends_on("mbedtls@2:3.5", when="@7.79:")
```

Comment:
This is incorrect. I prefer this style with separate bounds:

```python
depends_on("mbedtls@:2", when="@:7.78") # forward compat
depends_on("mbedtls@2:", when="@7.78:") # backward compat
```

since it is easier to comment why a forward compat bound is necessary as well.

Reply:
Okay dropped the only restrictive `depends_on` statement. Let me know what you think.